### ISSUE TITLE
False Positive S2392: variable read and written inside loop

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#712]
```javascript
function mapPPByLine(cfgCode) {
  let result = {};
  let currentBlock;  // FP "Move the declaration of "currentBlock" to line X." (inside loop)
  let lines = cfgCode.split('\n');
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    if (line.startsWith('B') && !line.startsWith('B0')) {
      currentBlock = line;
    } else if (line.startsWith('B0')) {
      result['B0.0'] = i;
    } else if (line.match(/^\d/)) {
      let blockElement = currentBlock + '.' + line.split(':')[0];
      result[blockElement] = i;
    }
  }
  return result;
}
```
